src/openeo_gfmap/fetching/generic.py

```diff
@@ -106,14 +106,24 @@ def generic_default_processor(cube: openeo.DataCube, **params):
     return generic_default_processor
 
 
+def _unavailable_collection_error(collection_name: str, _: FetchType):
+    raise ValueError(
+        f"Collection {collection_name} is not available for the selected backend."
+    )
+
+
 OTHER_BACKEND_MAP = {
     "AGERA5": {
         Backend.TERRASCOPE: {
             "fetch": partial(_get_generic_fetcher, collection_name="AGERA5"),
             "preprocessor": partial(_get_generic_processor, collection_name="AGERA5"),
         },
         Backend.CDSE: {
-            "fetch": partial(_get_generic_fetcher, collection_name="AGERA5"),
+            "fetch": _unavailable_collection_error,
```

Comment:
I do not think this is a sustainable implementation. In time if we port AGERA5 to CDSE, we will still get an error message in GFMAP. 

Ideally you do a try and a catch?

Reply:
Indeed, and I removed the backend map dictionnary because the generic fetcher and processor are used in all cases.